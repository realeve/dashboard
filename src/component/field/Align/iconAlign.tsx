import React from 'react';
import Icon from '@ant-design/icons';

const LeftSvg = () => (
  <svg width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false">
    <svg viewBox="0 0 1024 1024">
      <path d="M181.312 960H106.688V64h74.624v896zM928 773.312H256V250.688h672v522.624z m-74.688-74.624V325.312H330.688v373.376h522.624z"></path>
    </svg>
  </svg>
);

const CenterSvg = () => (
  <svg width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false">
    <svg viewBox="0 0 1024 1024">
      <path d="M160 250.688h312.832V64h78.272v186.688H864v522.624l-312.896 0.064V960H472.96l-0.064-186.624H160V250.624z m391.104 74.688v373.312h234.688V325.312l-234.688 0.064z m-312.896 0v373.312h234.624V325.376H238.208z"></path>
    </svg>
  </svg>
);

const RightSvg = () => (
  <svg width="1em" height="1em" fill="currentColor" aria-hidden="true" focusable="false">
    <svg viewBox="0 0 1024 1024">
      <path d="M852.352 64H928v896h-75.648V64zM96 250.688h680.704v522.624H96V250.688z m75.648 74.624v373.376h529.472V325.312H171.648z"></path>
    </svg>
  </svg>
);

export const AlignLeftIcon = (props) => <Icon component={LeftSvg} {...props} />;
export const AlignCenterIcon = (props) => <Icon component={CenterSvg} {...props} />;
export const AlignRightIcon = (props) => <Icon component={RightSvg} {...props} />;
