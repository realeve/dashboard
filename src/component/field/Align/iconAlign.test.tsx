import React from 'react';
import { create } from 'react-test-renderer';
import { AlignLeftIcon, AlignCenterIcon, AlignRightIcon } from './iconAlign';
test('文本对齐图标', () => {
  let root = create(<AlignLeftIcon />);

  expect(root.toJSON()).toMatchSnapshot();

  root.update(<AlignCenterIcon />);
  expect(root.toJSON()).toMatchSnapshot();

  root.update(<AlignRightIcon />);
  expect(root.toJSON()).toMatchSnapshot();
});
