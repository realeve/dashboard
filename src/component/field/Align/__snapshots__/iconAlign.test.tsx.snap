// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`文本对齐图标 1`] = `
<span
  className="anticon"
  role="img"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    focusable="false"
    height="1em"
    width="1em"
  >
    <svg
      viewBox="0 0 1024 1024"
    >
      <path
        d="M181.312 960H106.688V64h74.624v896zM928 773.312H256V250.688h672v522.624z m-74.688-74.624V325.312H330.688v373.376h522.624z"
      />
    </svg>
  </svg>
</span>
`;

exports[`文本对齐图标 2`] = `
<span
  className="anticon"
  role="img"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    focusable="false"
    height="1em"
    width="1em"
  >
    <svg
      viewBox="0 0 1024 1024"
    >
      <path
        d="M160 250.688h312.832V64h78.272v186.688H864v522.624l-312.896 0.064V960H472.96l-0.064-186.624H160V250.624z m391.104 74.688v373.312h234.688V325.312l-234.688 0.064z m-312.896 0v373.312h234.624V325.376H238.208z"
      />
    </svg>
  </svg>
</span>
`;

exports[`文本对齐图标 3`] = `
<span
  className="anticon"
  role="img"
>
  <svg
    aria-hidden="true"
    fill="currentColor"
    focusable="false"
    height="1em"
    width="1em"
  >
    <svg
      viewBox="0 0 1024 1024"
    >
      <path
        d="M852.352 64H928v896h-75.648V64zM96 250.688h680.704v522.624H96V250.688z m75.648 74.624v373.376h529.472V325.312H171.648z"
      />
    </svg>
  </svg>
</span>
`;
